konnectintech/Konectin-website-_backend
# Add Notification Preferences to User Schema and Implement Related Endpoints

# Description
This pull request introduces an addition to the user schema by adding notification preferences. Additionally, it implements a new endpoint to manage these preferences.

## Changes Made

### 1. Schema Update:
   - Update `notifications` of the user schema to store more notification preferences as boolean values.
   - Included fields for `resumeStatusUpdates`, `jobAlerts`, `internshipAlerts`, `blogUpdates`, and `reminders`.
   - Defaulted all notification preferences to `true` for new users.

### 2. Controller Implementation
   - Update function `updateNotificationSettings` to handle updating user notification preferences.
   - Implemented logic to find the user by ID, update the notification preferences based on the request body, and save the updated user.
   - Added error handling to handle cases such as user not found and server errors.

### 3. Routes Update
   - Created a new Put endpoint `/v2/updateNotificationSettings` to handle updating notification preferences for a specific user.
   - Linked the new endpoint to the `updateSettings` controller function.

## How  to Test

### 1. User Registration
   - Send a POST request to `http://localhost:3001/user/register` with the user's full name, email, and password in the request body.
   - Verify that the response indicates successful user creation.

### 2. Testing Notification Preferences Update
   - Send a PUT request to `http://localhost:3001/user//v2/updateNotificationSettings` with the user's ID  and the notification preferences you want to update in the request body.
   - For example, to update `jobAlerts` to `false`, include the following JSON object in the body:
     ```json
     {
         "jobAlerts": false
     }
     ```
   - Verify that the response indicates successful update of notification preferences and includes the updated user object.

The testing medium used is Postman.
